### PR TITLE
S0073 masked modulus10 1

### PR DESCRIPTION
**Pull request by KnowledgeForwardSolutions**
This pull request adds support for masked validation to the `Modulus10_1Algorithm`, allowing it to handle inputs with formatting characters (like spaces or dashes) via the `IMaskedCheckDigitAlgorithm` interface. It also introduces comprehensive unit tests for this new functionality and updates documentation and benchmarks accordingly.

**Support for masked validation in Modulus10_1Algorithm:**

* `Modulus10_1Algorithm` now implements `IMaskedCheckDigitAlgorithm`, enabling validation of formatted values using a mask. A new `Validate(String value, ICheckDigitMask mask)` method has been added, with logic to handle masked and unmasked digits and edge cases. [[1]](diffhunk://#diff-6cc6d05450f31a7a96060e3a8088e4595fc45cf451cb01c5583cc90a9bf34e00L23-R28) [[2]](diffhunk://#diff-6cc6d05450f31a7a96060e3a8088e4595fc45cf451cb01c5583cc90a9bf34e00R92-R136)
* The algorithm is now available via the `MaskedAlgorithms.Modulus10_1` property, and is documented as such in both code and the `README.md`. [[1]](diffhunk://#diff-ff964210b26c6fa4f14ae25a56e4cd7a73e5a962a77999317c4b0794acab6711R17-R19) [[2]](diffhunk://#diff-ff964210b26c6fa4f14ae25a56e4cd7a73e5a962a77999317c4b0794acab6711R51-R55) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R332) [[4]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R1316-R1319)

**Unit test enhancements:**

* Extensive new tests for the masked validation logic in `Modulus10_1Algorithm`, covering null/empty input, masking edge cases, correct weighting, and error detection.
* Additional unit tests for unmasked validation, including new benchmark values and explicit checks for non-digit characters and edge cases. [[1]](diffhunk://#diff-aacf76a9b4a051c89a954b8f2f6a17661a3c107bf682669bd17339c3ffa3f66dR50-R57) [[2]](diffhunk://#diff-aacf76a9b4a051c89a954b8f2f6a17661a3c107bf682669bd17339c3ffa3f66dL99-R128)
* Minor test improvements for `Modulus10_2Algorithm` to match the new coverage style.
* Tests for the new `MaskedAlgorithms.Modulus10_1` property.

**Benchmark and code quality updates:**

* Benchmark data updated to include `Modulus10_1` and comment out previous `Modulus10_2` entries, reflecting the new focus.
* Improved input validation and digit checking logic in both `Modulus10_1Algorithm` and related algorithms, using `IsInvalidDigit()` for clarity and correctness. [[1]](diffhunk://#diff-6cc6d05450f31a7a96060e3a8088e4595fc45cf451cb01c5583cc90a9bf34e00L49-R50) [[2]](diffhunk://#diff-6cc6d05450f31a7a96060e3a8088e4595fc45cf451cb01c5583cc90a9bf34e00R74-R82)
* Added clarifying comments and consistent accumulator logic across algorithms for maintainability and readability. [[1]](diffhunk://#diff-2001ac93c31d79f5213e87f07de88e5d3851da9de8392bf8949c8882c77a213fR45-R46) [[2]](diffhunk://#diff-f73a4dc1021aa09f507f982e0ee82b05f8d0e7c9c5bae0b9551504df2efd6d91R62-R63)

**Test setup improvements:**

* Test classes now instantiate and use relevant mask objects for more robust and clear test coverage. [[1]](diffhunk://#diff-aacf76a9b4a051c89a954b8f2f6a17661a3c107bf682669bd17339c3ffa3f66dR6-R8) [[2]](diffhunk://#diff-e81a83381aa378b88f05a25a10a04610885cb7e27c038caf35944d2524e4a06dL9)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1312,6 +1313,10 @@ Wikipedia: https://en.wikipedia.org/wiki/Luhn_algorithm
 The Modulus10 algorithm uses modulus 10 and each digit is weighted by its position
 in the value, starting with weight 1 for the right-most non-check digit character.
 
+`Modulus10_1Algorithm` implements [IMaskedCheckDigitAlgorithm](#imaskedcheckdigitalgorithm) and can be used
```
[the remark this answers is not in the document]
Comment (KnowledgeForwardSolutions) [reply README.md]: fixed
